runtime/doc/pi_health.txt

```diff
@@ -23,60 +23,135 @@ Commands				*health-commands*
 					*:checkhealth* *:CheckHealth*
 :checkhealth          Run all healthchecks.
 					*E5009*
-                      Nvim depends on |$VIMRUNTIME| and 'runtimepath' to find
-                      the standard "runtime files" for syntax highlighting,
-                      filetype-specific behavior, and standard plugins
-                      (including :checkhealth).  If the runtime files cannot
-                      be found then those features will not work.
+		Nvim depends on |$VIMRUNTIME| and 'runtimepath' to find the
+		standard "runtime files" for syntax highlighting,
+		filetype-specific behavior, and standard plugins (including
+		:checkhealth).  If the runtime files cannot be found then
+		those features will not work.
 
 :checkhealth {plugins}
-                      Run healthcheck(s) for one or more plugins. E.g. to run
-                      only the standard Nvim healthcheck: >
-                        :checkhealth nvim
-<                      To run the healthchecks for the "foo" and "bar" plugins
-                      (assuming these plugins are on your 'runtimepath' and
-                      they have implemented health#foo#check() and
-                      health#bar#check(), respectively): >
-                        :checkhealth foo bar
+		Run healthcheck(s) for one or more plugins. E.g. to run only
+		the standard Nvim healthcheck: >
+			:checkhealth nvim
 <
+		To run the healthchecks for the "foo" and "bar" plugins
+		(assuming these plugins are on your 'runtimepath' and they
+		have implemented the Lua or Vimscript interface
+		require("foo.health").check() and health#bar#check(),
+		respectively): >
+			:checkhealth foo bar
+<
+		To run healthchecks for lua submodules, use dot notation or
+		"*" to refer to all submodules. For example nvim provides
+		`vim.lsp` and `vim.treesitter`  >
+			:checkhealth vim.lsp vim.treesitter
+			:checkhealth vim*
+<
+==============================================================================
+Lua Functions				*health-functions-lua* *health-lua*
+
+health.vim Lua functions are for creating new healthchecks. They mostly just
+do some layout and formatting, to give users a consistent presentation. See
+also |health-functions-vim|
+
+The Lua health module provides the functions:  >
+	local health = require('health')
+<
```

Comment:
I don't think you need the "mostly just do some layout and formatting" part. If the role of the functions changes in the future then this becomes stale.

```suggestion
The Lua "health" module can be used to create new healthchecks (see also
|health-functions-vim|). To get started, simply use:  >
	local health = require('health')
<
```